lib/storage/storage.go

```diff
@@ -1157,6 +1157,8 @@ type PackageUpdate struct {
 	Labels map[string]string `json:"labels,omitempty"`
 	// ConfigPackage specifies optional configuration package dependency
 	ConfigPackage *PackageUpdate `json:"config_package,omitempty"`
+	// NoStart indicates that package service shouldn't be auto-started after upgrade
+	NoStart bool `json:"no_start"`
```

Comment:
I've been thinking about this flag - it feels bolted on with a very narrow purpose.
My suggestion is to back up a bit and consider moving this logic to [Config](https://github.com/gravitational/gravity/blob/version/5.5.x/lib/update/system/system.go#L168). Since I don't think the flag is useful for other services just yet, let's clean up the updater a bit:
```go
// Config defines the configuration of the system updater
type Config struct {
	// ... other fields omitted
	// NoPlanetAutostart specifies whether to avoid starting the planet container after installing
	// the service
	NoPlanetAutostart bool
}
```
As most of the `system.Update` invocations default to `true` for withStatus - this fits the behavioral pattern.
Then, `system.Update` becomes (no need for withStatus anymore):

```go
func (r *System) Update(ctx context.Context) error {
	// ...
	if r.NoPlanetAutostart {
		r.WithField("changeset", changeset).Info("System successfully updated.")
		return nil
	}
	// ...
}
```

and `updatePlanetPackage`:

```go
func (r *System) updatePlanetPackage(update storage.PackageUpdate) (labelUpdates []pack.LabelUpdate, err error) {
	// ...
	labelUpdates, err = r.reinstallService(update, r.NoPlanetAutostart)
	// ...
}
```

Reply:
Just thinking out loud, what about never auto starting and having the start always happen explicitly?

Reply:
@a-palchikov I don't know, I honestly don't see how moving the flag there is better or will make things cleaner. Right now it's nicely propagated everywhere it's needed via `PackageUpdate` (and IMO fits semantically there too). It serves a narrow purpose, yes, but `NoPlanetAutostart` is even more narrow. I can also anticipate a case where the upgrade would need to perform some actions after stopping but before starting Teleport service. So I'd prefer to keep it here if you don't feel too strongly about it.

@knisbet That's a valid idea and something I was exploring too, but I didn't want to mess with system upgrade logic too much in this PR.